src/rate_limiter.cc

```diff
@@ -274,10 +350,31 @@ RateLimiter::DequeuePayload(
   (*payload)->Callback();
   if ((*payload)->GetInstance() == nullptr) {
     (*payload)->SetInstance(instances.front());
+    // Enqueue did not specify the specific instance to
+    // run with the payload. Hence, need to explicitly
+    // decrement the consumer count for the instance
+    // which got allocated.
+    payload_queue->specific_queues_[instances.front()]
+        ->DecrementConsumerCount();
     instances.pop_front();
   } else {
     instances.erase(instances.begin() + instance_index);
   }
+
+  // Decrement the counts from the remaining specific
+  // instance handling as there will be no consumer for
+  // these queues.
+  // FIXME: For more accurate handling, these instances
```

Comment:
Would you be able to create a JIRA ticket for this and link it here? Ideally with more details. I'm not understanding from the comment what the concern about the current approach is, the costs (versus the other one), and why we're using an approximation here, so getting that information into the JIRA ticket would be good.